generator/api/src/main/java/io/fabric8/maven/generator/api/support/AbstractPortsExtractor.java

```diff
@@ -0,0 +1,178 @@
+
+    private static final String NUMBER_REGEX = "\\d+";
+
+    private static final String PORT_REGEX = "([a-zA-Z0-9_]+)(([\\.-_]+p{1})|([P]{1}))ort";
+    private static final Pattern PORT_PATTERN = Pattern.compile(PORT_REGEX);
+
+    private final PrefixedLogger log;
+
+    public AbstractPortsExtractor(PrefixedLogger log) {
+        this.log = log;
+    }
+
+    /**
+     * @return The name of the system property that points to the path of the configuration file.
+     */
+    public abstract String getConfigPathPropertyName();
+
+    /**
+     * Finds the name of the configuration file from the {@link MavenProject}.
+     * @param project   The {@link MavenProject} to use.
+     * @return          The path to the configuration file.
+     */
+    public abstract String getConfigPathFromProject(MavenProject project);
+
+
+    public File getConfigLocation(MavenProject project) {
+        //The system property has priority over what is specified in the pom.
+        String configPath = System.getProperty(getConfigPathPropertyName(), getConfigPathFromProject(project));
+        if (configPath == null) {
+            return null;
+        }
+        return Paths.get(configPath).toFile();
+    }
+
+    @Override
+    public Map<String, Integer> extract(MavenProject project) {
+        Map<String, Integer> answer = new HashMap<>();
+        File configFile = getConfigLocation(project);
+        if (!configFile.exists()) {
+            log.warn("Could not find config: [%s]. Ignoring.", configFile.getAbsolutePath());
+            return null;
+        }
+
+        try {
+            Map<String, String> configMap = readConfig(configFile);
+            for (Map.Entry<String, String> entry : configMap.entrySet()) {
+                String key = entry.getKey();
+                if (isValidPortPropertyKey(key)) {
+                    addPortIfValid(answer, key, entry.getValue());
+                }
+            }
+        } catch (IOException e) {
+            log.warn("Error reading config: [%s], due to: [%s]. Ignoring.", configFile.getAbsolutePath(), e.getMessage());
+        }
+        return answer;
+    }
+
+
+
+    /**
+     * Reads the configuration from the file.
+     * @param f
+     * @return
+     * @throws IOException
+     */
+    static Map<String, String> readConfig(File f) throws IOException {
+        Map<String, String> map;
+        if (f.getName().endsWith(JSON_EXTENSION)) {
+            map = flatten(JSON_MAPPER.readValue(f, Map.class));
+        } else if (f.getName().endsWith(YAML_EXTENSION) || f.getName().endsWith(YML_EXTENSION)) {
+            map = flatten(YAML_MAPPER.readValue(f, Map.class));
+        } else if (f.getName().endsWith(PROPERTIES_EXTENSION)) {
+            Properties properties = new Properties();
+            properties.load(new FileInputStream(f));
+            map = propertiesToMap(properties);
+        } else {
+            map = Collections.emptyMap();
```

Comment:
Maybe better to throw an exception here, that the config format (like 'toml' as Vert.x is starting to look in) is not supported, instead of silently failing to extract a port from a given file name.